lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1204,41 +1204,96 @@ def cmd_route_tabs(str, words)
   end
 
   def cmd_save_help
-    print_line "Usage: save"
+    print_line "Usage:"
     print_line
-    print_line "Save the active datastore contents to disk for automatic use across restarts of the console"
+    print_line "save:                       save the active module options to disk for automatic use across restarts of the console"
+    print_line "save -d / --delete:         delete the active module options from disk without refreshing current options"
+    print_line "save -r / --reload:         reload default options for the active module"
+    print_line "save -l / --load:           load the saved options for the active module"
+    print_line "save -D / --delete-all:     delete ALL datastore options from the config file"
     print_line
     print_line "The configuration is stored in #{Msf::Config.config_file}"
     print_line
   end
 
   #
-  # Saves the active datastore contents to disk for automatic use across
+  # Deletes or saves the active datastore contents to disk for automatic use across
   # restarts of the console.
   #
   def cmd_save(*args)
-    # Save the console config
-    driver.save_config
-
-    begin
-      FeatureManager.instance.save_config
-    rescue StandardException => e
-      elog(e)
+    case args[0]
```

Comment:
Some of the commands use Rex::Parser::Arguments but it doesn't seem entirely consistent, I think it might help tidy up some boilerplate code you have here though 
https://github.com/rapid7/metasploit-framework/blob/d0e4d15d5348a721aa3234e8069b202dfe1f9c40/lib/msf/ui/console/command_dispatcher/core.rb#L65

https://github.com/rapid7/metasploit-framework/blob/d0e4d15d5348a721aa3234e8069b202dfe1f9c40/lib/msf/ui/console/command_dispatcher/core.rb#L302

Reply:
Thanks! I've changed the previous pattern over to this as of the latest commit. 👍